### ISSUE TITLE
ExplicitImplicitTypes: don't warn in FunctionN

### ISSUE THREAD

**Issue by xuwei-k** [xuwei-k/scalafix-rules#21]
```scala
Welcome to Scala 3.1.1 (1.8.0_322, Java OpenJDK 64-Bit Server VM).
Type in expressions for evaluation. Or try :help.
                                                                                                                                                                                                                                                                                                    
scala> val f: Int => Int = { a =>
     |   implicit val s = "a"
     |   a
     | }
val f: Int => Int = Lambda$1227/264138968@3b4086c1
```
